### PR TITLE
Audioviz

### PR DESCRIPTION
**Pull request by takecare**
Provides two separate classes to listen in to audio input from the microphone. Makes this data available to whoever wants to listen in. Sample test draws a rectangle based on one input source.

soundview

### CODE REVIEW

File: mobile/src/androidTest/java/com/novoda/canvas/audioviz/SimpleSoundMeter.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            audioRecord = new AudioRecord(
+                    MediaRecorder.AudioSource.MIC,
+                    SAMPLE_RATE,
+                    AudioFormat.CHANNEL_IN_MONO,
+                    AudioFormat.ENCODING_PCM_16BIT,
+                    minBufferSize
+            );
+        }
+
+        audioRecord.startRecording();
```
Comment (friedger) [inline mobile/src/androidTest/java/com/novoda/canvas/audioviz/SimpleSoundMeter.java]: On the Nexus 9, I get 
```
10-29 12:51:06.922 13273-13273/com.novoda.canvas E/AndroidRuntime: java.lang.IllegalStateException: startRecording() called on an uninitialized AudioRecord.
10-29 12:51:06.922 13273-13273/com.novoda.canvas E/AndroidRuntime:     at android.media.AudioRecord.startRecording(AudioRecord.java:885)
10-29 12:51:06.922 13273-13273/com.novoda.canvas E/AndroidRuntime:     at com.novoda.canvas.audioviz.SimpleSoundMeter.start(SimpleSoundMeter.java:36)
10-29 12:51:06.922 13273-13273/com.novoda.canvas E/AndroidRuntime:     at com.novoda.canvas.audioviz.MeanSoundViewTest.setupDependencies(MeanSoundViewTest.java:23)
10-29 12:51:06.922 13273-13273/com.novoda.canvas E/AndroidRuntime:     at com.novoda.canvas.audioviz.MeanSoundViewTest.startTestFor(MeanSoundViewTest.java:30)
10-29 12:51:06.922 13273-13273/com.novoda.canvas E/AndroidRuntime:     at com.novoda.canvas.base.NovodaActivityTest$1.run(NovodaActivityTest.java:40)
10-29 12:51:06.922 13273-13273/com.novoda.canvas E/AndroidRuntime:     at android.os.Handler.handleCallback(Handler.java:739)
10-29 12:51:06.922 13273-13273/com.novoda.canvas E/AndroidRuntime:     at android.os.Handler.dispatchMessage(Handler.java:95)
10-29 12:51:06.922 13273-13273/com.novoda.canvas E/AndroidRuntime:     at android.os.Looper.loop(Looper.java:148)
10-29 12:51:06.922 13273-13273/com.novoda.canvas E/AndroidRuntime:     at android.app.ActivityThread.main(ActivityThread.java:5417)
10-29 12:51:06.922 13273-13273/com.novoda.canvas E/AndroidRuntime:     at java.lang.reflect.Method.invoke(Native Method)
```

Comment (takecare) [inline mobile/src/androidTest/java/com/novoda/canvas/audioviz/SimpleSoundMeter.java]: will check asap

**Comment by blundell** [comment]
Gif?

**Comment by takecare** [comment]
@blundell giant gif in place

**Comment by friedger** [comment]
LGTM

**Comment by ouchadam** [comment]
let's do it!

**Comment by blundell** [comment]
:+1:

**Pull request merged**